### ISSUE TITLE
Error when serializing enum with struct

### ISSUE THREAD

**Issue by Shatur** [toml-rs/toml#329]
Consider the following code:

```rust
use serde::{Deserialize, Serialize};

#[derive(Deserialize, Serialize)]
struct Foo {
    bar: Baz,
}

#[derive(Deserialize, Serialize)]
enum Baz {
    Test(Test)
}

#[derive(Deserialize, Serialize)]
struct Test {
    name: String,
}

fn main() {
    let string = std::fs::read_to_string("test.toml").unwrap();
    let foo: Foo = toml_edit::de::from_str(&string).unwrap();
    toml_edit::ser::to_string(&foo).unwrap();
}
```
with the following `test.toml`:
```toml
bar = { Test = { name = "asdf" } }
```

Deserialization works, but serialization panics with
```
called `Result::unwrap()` on an `Err` value: Error { kind: UnsupportedType }
```

**Comment by epage** [comment]
I wonder if #321 is related

**Comment by epage** [comment]
#405 might also be related

**Comment by epage** [comment]
This was fixed in #557 which has a sufficient test for this case.

Thanks for checking!
